src/components/GenerateTrackingCodeModal.vue

```diff
@@ -0,0 +1,284 @@
+          <ion-select-option v-for="carrier in facilityCarriers" :key="carrier.partyId" :value="carrier.partyId">{{ translate(carrier.groupName) }}</ion-select-option>
+        </ion-select>
+      </ion-item>
+      <ion-item>
+        <ion-select :disabled="!order.missingLabelImage" :label="translate('Method')" v-model="shipmentMethodTypeId" interface="popover">
+          <ion-select-option v-for="method in carrierMethods" :key="method.productStoreShipMethId" :value="method.shipmentMethodTypeId">{{ translate(method.description) }}</ion-select-option>
+        </ion-select>
+      </ion-item>
+      <ion-item>
+        <ion-input :label="translate('Tracking code')" :placeholder="translate('enter code')" v-model="trackingCode" />
+      </ion-item>
+      <ion-item>
+        <ion-label>{{ translate("Tracking URL:", { trackingUrl: generateTrackingUrl() }) }}</ion-label>
+        <ion-button slot="end" fill="clear" size="default" :disabled="!trackingCode.trim() || !getCarrierTrackingUrl()" @click="redirectToTrackingUrl()">
+          {{ translate("Test") }}
+          <ion-icon :icon="openOutline" slot="end" />
+        </ion-button>
+      </ion-item>
+    </ion-list>
+  </ion-content>
+
+  <ion-fab vertical="bottom" horizontal="end" slot="fixed">
+    <ion-fab-button :disabled="isTrackingRequired ? !shipmentMethodTypeId: ''" @click="confirmSave()">
+      <ion-icon :icon="isForceScanEnabled ? barcodeOutline : saveOutline" />
+    </ion-fab-button>
+  </ion-fab>
+</template>
+
+<script lang="ts">
+import {
+  IonButton,
+  IonButtons,
+  IonContent,
+  IonFab,
+  IonFabButton,
+  IonHeader,
+  IonIcon,
+  IonInput,
+  IonItem,
+  IonLabel,
+  IonList,
+  IonSelect,
+  IonSelectOption,
+  IonTitle,
+  IonToolbar,
+  modalController,
+} from "@ionic/vue";
+import { defineComponent } from "vue";
+import { barcodeOutline, closeOutline, copyOutline, openOutline, saveOutline } from "ionicons/icons";
+import { translate } from "@hotwax/dxp-components";
+import { mapGetters, useStore } from "vuex";
+import { OrderService } from '@/services/OrderService';
+import logger from "@/logger";
+import { showToast } from "@/utils";
+import { hasError } from "@/adapter";
+
+export default defineComponent({
+  name: "GenerateTrackingCodeModal",
+  components: { 
+    IonButton,
+    IonButtons,
+    IonContent,
+    IonFab,
+    IonFabButton,
+    IonHeader,
+    IonIcon,
+    IonInput,
+    IonItem,
+    IonLabel,
+    IonList,
+    IonSelect,
+    IonSelectOption,
+    IonTitle,
+    IonToolbar,
+  },
+  computed: {
+    ...mapGetters({
+      facilityCarriers: 'carrier/getFacilityCarriers',
+      isForceScanEnabled: 'util/isForceScanEnabled',
+      productStoreShipmentMethods: 'carrier/getProductStoreShipmentMethods',
+      productStoreShipmentMethCount: 'util/getProductStoreShipmentMethCount',
+      completedOrders: 'order/getCompletedOrders',
```

Comment:
Do we really need to fetch the completedOrders here?

Reply:
Ther is no need, Removed this from the PR.